managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1868,7 +1876,12 @@ public void asyncMarkDelete(final Position position, Map<String, Long> propertie
 
         if (config.isDeletionAtBatchIndexLevelEnabled() && batchDeletedIndexes != null) {
             if (newPosition.ackSet != null) {
-                batchDeletedIndexes.put(newPosition, BitSetRecyclable.create().resetWords(newPosition.ackSet));
+                lock.writeLock().lock();
+                try {
+                    setBatchDeletedIndexes(newPosition);
```

Comment:
Looks like we just change to 

```java
BitSetRecyclable givenBitSet = BitSetRecyclable.create().resetWords(newPosition.ackSet);
batchDeletedIndexes.compute(newPosition,
        (k, v) -> v == null || givenBitSet.nextClearBit(0) > v.nextClearBit(0) ?
                givenBitSet : v);
```

Reply:
OK ,I will fix

Reply:
Fixed, PTAL,thanks! @codelipenghui

Reply:
nextClearBit is not suitable, because 0 means acked.
So use nextSetBit instead.